src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceService.java

```diff
@@ -572,6 +578,52 @@ public void enableIamPermissions(Map<String, List<String>> userPermissions, Stri
         throw new EnablePermissionsFailedException("Cannot enable iam permissions", lastException);
     }
 
+    public void revokeIamPermissions(Map<String, List<String>> userPermissions, String projectId)
+        throws InterruptedException {
+
+        GetIamPolicyRequest getIamPolicyRequest = new GetIamPolicyRequest();
+
+        Exception lastException = null;
+        int retryWait = INITIAL_WAIT_SECONDS;
+        for (int i = 0; i < RETRIES; i++) {
+            try {
+                CloudResourceManager resourceManager = cloudResourceManager();
+                Policy policy = resourceManager.projects()
+                    .getIamPolicy(projectId, getIamPolicyRequest).execute();
+                final List<Binding> bindingsList = policy.getBindings();
+
+                // Remove members from the current policies
+                for (Map.Entry<String, List<String>> entry : userPermissions.entrySet()) {
+                    bindingsList.forEach(b -> {
+                        if (Objects.equals(b.getRole(), entry.getKey())) {
+                            // Remove the members that were passed in
+                            b.setMembers(ListUtils.subtract(b.getMembers(), entry.getValue()));
+                        }
+                    });
+                }
+
+                // Remove any entries from the bindings list with no members
+                CollectionUtils.filter(bindingsList, b -> !b.getMembers().isEmpty());
+
+                policy.setBindings(bindingsList);
+                SetIamPolicyRequest setIamPolicyRequest = new SetIamPolicyRequest().setPolicy(policy);
+                resourceManager.projects()
+                    .setIamPolicy(projectId, setIamPolicyRequest).execute();
+                return;
+            } catch (IOException | GeneralSecurityException ex) {
+                logger.info("Failed to revoke iam permissions. Retry " + i + " of " + RETRIES, ex);
+                lastException = ex;
+            }
+
+            TimeUnit.SECONDS.sleep(retryWait);
+            retryWait = retryWait + retryWait;
+            if (retryWait > MAX_WAIT_SECONDS) {
+                retryWait = MAX_WAIT_SECONDS;
+            }
+        }
+        throw new RevokePermissionsFailedException("Cannot revoke iam permissions", lastException);
```

Comment:
If retries are exceeded then `lastException` will be `null` when this line is reached. Maybe
```
if (lastException != null) {
    throw new RevokePermissionsFailedException("Cannot revoke iam permissions", lastException);
} else {
    throw new RevokePermissionsFailedException("Cannot revoke iam permissions: too many retries");
}
```

Reply:
I think it's getting set on line 615

Reply:
Oh, of course, the retry only occurs if an exception was thrown, duh